lib/skate/warm_up.ex

```diff
@@ -0,0 +1,92 @@
+defmodule Skate.WarmUp do
+  use GenServer
+  require Logger
+
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, :ok)
+  end
+
+  @impl true
+  def init(state) do
+    pool_size = Application.get_env(:skate, Skate.Repo)[:pool_size]
+
+    minimum_percent_queries_to_succeed =
+      Application.get_env(:skate, Skate.WarmUp)[:minimum_percent_queries_to_succeed] || 1
+
+    max_attempts = Application.get_env(:skate, Skate.WarmUp)[:max_attempts]
+
+    warm_up_check_fn =
+      Application.get_env(:skate, :warm_up_test_fn, fn _index, _attemptgit ->
+        Skate.Repo.query("SELECT 1", [])
+      end)
+
+    case check_connections_warmed_up(%{
+           pool_size: pool_size,
+           required_success_count: pool_size * minimum_percent_queries_to_succeed,
+           max_attempts: max_attempts,
+           attempt: 1,
+           check_fn: warm_up_check_fn
+         }) do
+      %{status: :success} = result ->
+        Logger.info(format_result_message(result))
+        {:ok, state}
+
+      %{status: :failure} = result ->
+        message = format_result_message(result)
+        Logger.error(message)
+        {:stop, message}
+    end
+  end
+
+  defp check_connections_warmed_up(
+         %{
+           pool_size: pool_size,
+           required_success_count: required_success_count,
+           max_attempts: max_attempts,
+           attempt: attempt,
+           check_fn: check_fn
+         } = config
+       ) do
+    count_success =
+      Enum.map(0..(pool_size - 1), fn index ->
```

Comment:
running the same number of queries as there are connections in the pool isn't really an exact science, but gives some level of confidence that multiple concurrent queries should be able to succeed quickly